### PR TITLE
Add vector service endpoints

### PR DESCRIPTION
**Pull request by mattnibs**

### CODE REVIEW

File: api/client/connection.go [hunk 1]
```diff
@@ -432,6 +432,25 @@ func (c *Connection) Vacuum(ctx context.Context, pool, revision string, dryrun b
 	return res, err
 }
 
+func (c *Connection) VectorPost(ctx context.Context, pool, revision string, objectIDs []ksuid.KSUID, message api.CommitMessage) (api.CommitResponse, error) {
+	return c.vector(ctx, pool, revision, objectIDs, message, http.MethodPost)
+}
+
+func (c *Connection) VectorDelete(ctx context.Context, pool, revision string, objectIDs []ksuid.KSUID, message api.CommitMessage) (api.CommitResponse, error) {
```
Comment (nwt) [inline api/client/connection.go:439]: Call these AddVectors and DeleteVectors so they match the lake/api.Interface? That's what we've done with the all the other methods.

File: api/client/connection.go [hunk 2]
```diff
@@ -432,6 +432,25 @@ func (c *Connection) Vacuum(ctx context.Context, pool, revision string, dryrun b
...
+	return c.vector(ctx, pool, revision, objectIDs, message, http.MethodDelete)
+}
+
+func (c *Connection) vector(ctx context.Context, pool, revision string, objectIDs []ksuid.KSUID, message api.CommitMessage, method string) (api.CommitResponse, error) {
```
Comment (nwt) [inline api/client/connection.go:443]: Nit: Call this doVector so the name is verby.

File: service/ztests/vector.yaml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  zed vector add -q $id
+  zed query -Z 'from POOL@main:vectors | drop id'
+  echo ===
+  zed vector delete -q $id
+  zed query -Z 'from POOL@main:vectors | drop id'
+  echo ===
+
+inputs:
+  - name: service.sh
+    source: service.sh
```
Comment (nwt) [inline service/ztests/vector.yaml:15]: Nit: Don't need this.
```suggestion
```

File: service/ztests/vector.yaml [hunk 4]
```diff
@@ -0,0 +1,33 @@
+script: |
```
Comment (nwt) [inline service/ztests/vector.yaml:1]: We should add the local lake equivalent of this test in lake/ztests/vector.yaml so we have coverage for the lake/api/local.go change.

**Review by nwt — COMMENTED** [review]
We should add this to docs/lake/api.md.

File: service/ztests/vector.yaml (hunk 4 above)
Comment (mattnibs) [reply service/ztests/vector.yaml:1]: This test is an exact copy of lake/ztests/vector.yaml

Comment (nwt) [reply service/ztests/vector.yaml:1]: My apologies. I totally missed that.

**Review by nwt — APPROVED** [review]
I'm OK merging this without the docs/lake/api.md update if you'd rather do that later.

**Pull request merged**